lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -236,16 +237,19 @@ tokenize([$?, $\\, H | T], Line, Column, Scope, Tokens) ->
   tokenize(T, Line, Column + 3, Scope, [Token | Tokens]);
 
 tokenize([$?, Char | T], Line, Column, Scope, Tokens) ->
-  case handle_char(Char) of
+  NewScope = case handle_char(Char) of
     {Escape, Name} ->
       Msg = io_lib:format("found ? followed by code point 0x~.16B (~ts), please use ?~ts instead",
                           [Char, Name, Escape]),
-      elixir_errors:erl_warn(Line, Scope#elixir_tokenizer.file, Msg);
+      elixir_errors:erl_warn(Line, Scope#elixir_tokenizer.file, Msg),
+      Scope#elixir_tokenizer{warnings =
+        Scope#elixir_tokenizer.warnings ++ [{Line, Scope#elixir_tokenizer.file, Msg}]
```

Comment:
Instead of appending, consider prepending the warnings:

```elixir
         [{Line, Scope#elixir_tokenizer.file, Msg} | Scope#elixir_tokenizer.warnings]
```

Then when you return it, you do `lists:reverse(Warnings)`.

Also, it probably makes sense to have a function called `prepend_warning({Line, File, Msg}, Scope)` to reduce the duplication. For now, the `prepend_warning` function can call `elixir_errors:erl_warn(Line, Scope#elixir_tokenizer.file, Msg)` and prepend the warning. This way, once we have to stop emitting the warning, we need to do it only once.

Reply:
Sure. No need to reverse the warning list though, as you can get an ordered one and without the duplicated warnings by using `ordsets` instead of `sets`, as I was doing.